### PR TITLE
 Connecting CAN Frames to ROS, issue #326 

### LINKED ISSUE #326: Connect CAN Frames to ROS
### Purpose

Connect CAN Frame parsing logic to ROS via the CAN Transceiver

### Resources

-

### PR DESCRIPTION
**Pull request by lross03**
### Description

- Resolves #326 
- Implemented pairing of CAN frames with their respective ROS publishing functions.

### CODE REVIEW

**Comment by lross03** [comment]
In the filtered wind sensor publishing function, I was getting an error saying I was narrowing the type conversion from int to int16 when I was calculating the average direction. As far as I can tell, direction is an int16 in the custom interface, so I don't know why I was getting this error. I added a nolint (bugprone-narrowing-conversions) for now. 

I also wasn't sure how to approach finding the correct generic sensor from the generic sensor custom interface, as generic sensors are not in the can_frame_parser, so I used the same method as the battery publishing function and assumed that the sensor will be in the generic_sensors array from custom interfaces.

**Review by hhenry01 — CHANGES_REQUESTED** [review]
Please address the sections marked with `TODO(lross03)`

File: src/network_systems/projects/can_transceiver/src/can_frame_parser.cpp [hunk 1]
```diff
@@ -440,4 +440,200 @@ void GPS::checkBounds() const
...
+{
+    int32_t  raw_id;  // CAN documentation says id is uint32, but custom interfaces says its int32
+    uint32_t raw_lat;
+    uint32_t raw_lon;
+    uint16_t raw_speed;
+    uint16_t raw_course;
+    uint16_t raw_heading;
+    int8_t   raw_rot;
+    uint16_t raw_length;
+    uint8_t  raw_width;
```
Comment (samdai01) [inline src/network_systems/projects/can_transceiver/src/can_frame_parser.cpp:459]: According to `CAN Frames`, the width is of type `uint16_t`

**Review by samdai01 — CHANGES_REQUESTED** [review]
issue with data size of ship width

File: src/network_systems/projects/can_transceiver/src/can_frame_parser.cpp [hunk 2]
```diff
@@ -440,4 +440,200 @@ void GPS::checkBounds() const
...
+
+    std::memcpy(&raw_id, cf.data + BYTE_OFF_ID, sizeof(int32_t));
+    std::memcpy(&raw_lat, cf.data + BYTE_OFF_LAT, sizeof(int32_t));
+    std::memcpy(&raw_lon, cf.data + BYTE_OFF_LON, sizeof(int32_t));
+    std::memcpy(&raw_speed, cf.data + BYTE_OFF_SPEED, sizeof(int16_t));
+    std::memcpy(&raw_course, cf.data + BYTE_OFF_COURSE, sizeof(int16_t));
+    std::memcpy(&raw_heading, cf.data + BYTE_OFF_HEADING, sizeof(int16_t));
+    std::memcpy(&raw_rot, cf.data + BYTE_OFF_ROT, sizeof(int8_t));
+    std::memcpy(&raw_length, cf.data + BYTE_OFF_LENGTH, sizeof(int16_t));
+    std::memcpy(&raw_width, cf.data + BYTE_OFF_WIDTH, sizeof(int8_t));
```
Comment (samdai01) [inline src/network_systems/projects/can_transceiver/src/can_frame_parser.cpp:471]: Please change this to copy `uint16_t` to `raw_width`

**Review by samdai01 — APPROVED** [review]

**Pull request merged**